### ISSUE TITLE
GPU usage on iGPU is bugged 9000%

### ISSUE THREAD

**Issue by JaviBOT96** [flightlessmango/MangoHud#731]
Hello there. I am using Archlinux and this issue happens with every single game and emulator I use.
I have a Ryzen 5 5600G
16gb 3200mhz
Vega 7

![Screenshot_20220413_033520](https://user-images.githubusercontent.com/67564457/163135439-088aa5f5-49af-4b21-bb58-41fc8ebdaee4.png)

**Comment by JaviBOT96** [comment]
![image](https://user-images.githubusercontent.com/67564457/163145636-fdad9b65-cb60-4d72-97b9-4111c58d0499.png)
Thank you for replying.

**Comment by jackun** [comment]
v2.2, seems kernel/FW bug then, 6457% usage in screenshot (if i can math gud) :confused:

**Comment by JaviBOT96** [comment]
Oh, I do apologize. I didn't mention the kernel I am using. 
![image](https://user-images.githubusercontent.com/67564457/163147717-31c1a664-8d58-4fe2-af90-c265936cac0b.png)

**Comment by jackun** [comment]
If you're not already using latest git `master` branch, try building that.
Older versions use `/sys/class/drm/card*/device/gpu_busy_percent`, maybe it outputs garbage.

**Comment by KookiesGH** [comment]
I'm having a similar issue on my system. Using a Ryzen 5 PRO 4650U, 16 GB 3200MHz, with Renoir graphics.
![gpu-usage](https://user-images.githubusercontent.com/66989235/163659141-2b9624b5-5ff7-4534-a47f-bfd664222f7c.png)

Here is the output of `hexdump -C /sys/class/drm/card0/device/gpu_metrics`
```
00000000  80 00 02 02 6e 0f eb 0f  b9 0f a0 0f b9 0f b9 0f  |....n...........|
00000010  1d 10 b9 0f eb 0f 87 0f  a0 0f d2 0f 4a 03 00 00  |............J...|
00000020  79 cd 58 30 88 09 00 00  04 00 8a 05 99 03 ff ff  |y.X0............|
00000030  00 00 af 01 1f 00 1d 00  6a 01 44 00 8a 00 00 00  |........j.D.....|
00000040  22 02 b4 01 ff ff a6 04  90 01 ff ff 90 01 90 01  |"...............|
00000050  06 00 b0 04 90 01 90 01  00 00 78 05 78 05 78 05  |..........x.x.x.|
00000060  78 05 78 05 78 05 00 00  78 05 78 05 00 00 00 00  |x.x.x...x.x.....|
00000070  00 00 ff ff ff ff ff ff  00 00 00 00 00 00 00 00  |................|
00000080
```
And the output of `hexdump -C /sys/class/drm/card0/device/gpu_busy_percent`
```
00000000  38 0a                                             |8.|
00000002
```

**Comment by flightlessmango** [comment]
can anyone check if this is fixed in latest? 5349226fa50f98c7d3328258112f48865b96cddb

**Comment by KookiesGH** [comment]
Looks good to me. Thanks for the fix!
![image](https://user-images.githubusercontent.com/66989235/164913743-00de18df-a46e-4152-b4e9-d6cbd12048b4.png)

**Comment by flightlessmango** [comment]
Is that with version 0.6.7?

**Comment by VladimirMrzv** [comment]
with 0.6.7 
on 0.6.5 all normal

**Comment by flightlessmango** [comment]
@VladimirMrzv this commit should fix the gpu temp b30302c889eed2d01744411233a041a0491cfba9, can you verify?

**Comment by VladimirMrzv** [comment]
seems works
![image](https://user-images.githubusercontent.com/81020401/167134384-b27976ff-f1fc-40f4-a37b-50ae979c428d.png)

![image](https://user-images.githubusercontent.com/81020401/167134423-33b79562-1cb8-49fb-83a8-2fffdb7039e6.png)

in new version missed GPU name and mesa version

**Comment by VladimirMrzv** [comment]
> @VladimirMrzv this commit should fix the gpu temp [b30302c](https://github.com/flightlessmango/MangoHud/commit/b30302c889eed2d01744411233a041a0491cfba9), can you verify?

It shows wrong GPU usage. In my screenshots it  equal to temperature, real not more 1%

**Comment by flightlessmango** [comment]
There's been some more changes in this area, could you try build latest and see how it works?

**Comment by VladimirMrzv** [comment]
with new build
![image](https://user-images.githubusercontent.com/81020401/167632503-91134312-f764-45c7-a255-376b797f8b1b.png)

**Comment by flightlessmango** [comment]
@VladimirMrzv this commit should fix it 5167e72649a43478d8d96bf7a9c964934c72e5ba
